app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt

```diff
@@ -55,19 +55,23 @@ class MeetingRequestService : FirebaseMessagingService() {
    *    returns `false`.
    */
   fun isAppInForeground(): Boolean {
-    val activityManager = getSystemService(Context.ACTIVITY_SERVICE) as? ActivityManager
-    if (activityManager == null) {
-      Log.w("MeetingRequestService", "ActivityManager not available")
-      return false
-    }
-    val appProcesses = activityManager.runningAppProcesses ?: return false
-    val packageName = packageName
+    try {
+      val activityManager = getSystemService(Context.ACTIVITY_SERVICE) as? ActivityManager
+      if (activityManager == null) {
+        Log.w("MeetingRequestService", "ActivityManager not available")
+        return false
+      }
+      val appProcesses = activityManager.runningAppProcesses ?: return false
+      val packageName = packageName
 
-    for (appProcess in appProcesses) {
-      if (appProcess.importance == ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND &&
-          appProcess.processName == packageName) {
-        return true
+      for (appProcess in appProcesses) {
+        if (appProcess.importance == ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND &&
+            appProcess.processName == packageName) {
+          return true
+        }
       }
+    } catch (e: NullPointerException) {
+      Log.d("endToEnd", "Normal in testing mode but this should not happen in production.")
```

Comment:
Question: I don't really understand what does this do.

Reply:
In Andoid tests, the context inside the app is `null`, so the call to `Context.ACTIVITY_SERVICE` will throw a `NullPointerException`. Since in the end-to-end tests I don't test if I have received a notification, I decided to skip this edge case that way, and since the context should never be `null` in production this should be okay. There are many more elegant ways to do this but it is the easiest way.